src/main/java/uk/gov/gchq/palisade/resource/request/GetResourcesByTypeRequest.java

```diff
@@ -44,8 +45,8 @@ public GetResourcesByTypeRequest type(final String type) {
         return this;
     }
 
+    @Generated
     public String getType() {
-        requireNonNull(type, "The resource type has not been set.");
         return type;
     }
 
```

Comment:
setter requires @generated and null check

update equals, hashCode and toString methods

Reply:
set type requires @generated